core/src/test/scala/kafka/raft/KafkaMetadataLogTest.scala

```diff
@@ -196,6 +227,53 @@ final class KafkaMetadataLogTest {
     assertEquals(greaterEpochSnapshotId.offset, log.highWatermark.offset)
   }
 
+  @Test
+  def testTruncateWillRemoveOlderSnapshot(): Unit = {
+
+    val topicPartition = new TopicPartition("cluster-metadata", 0)
+    val (logDir, log) = buildMetadataLogAndDir(tempDir, mockTime, topicPartition)
+    val numberOfRecords = 10
+    val epoch = 1
+
+    append(log, 1, epoch - 1)
+    val oldSnapshotId1 = new OffsetAndEpoch(1, epoch - 1)
+    TestUtils.resource(log.createSnapshot(oldSnapshotId1)) { snapshot =>
+      snapshot.freeze()
+    }
+
+    append(log, 1, epoch, log.endOffset.offset)
+    val oldSnapshotId2 = new OffsetAndEpoch(2, epoch)
+    TestUtils.resource(log.createSnapshot(oldSnapshotId2)) { snapshot =>
+      snapshot.freeze()
+    }
+
+    append(log, numberOfRecords - 2, epoch, log.endOffset.offset)
+    val oldSnapshotId3 = new OffsetAndEpoch(numberOfRecords, epoch)
+    TestUtils.resource(log.createSnapshot(oldSnapshotId3)) { snapshot =>
+      snapshot.freeze()
+    }
+
+    val greaterEpochSnapshotId = new OffsetAndEpoch(3 * numberOfRecords, epoch)
```

Comment:
Minor but this snapshot has the same epoch as some of the other snapshots and the "last fetched epoch" of the log. Did you mean something like `greaterSnapshotId` or just `latestSnapshotId`?